chaoscenter/authentication/api/handlers/rest/project_handler.go

```diff
@@ -0,0 +1,679 @@
+)
+
+// GetUserWithProject returns user and project details based on username
+func GetUserWithProject(service services.ApplicationService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		username := c.Param("username")
+
+		// Validating logged in user
+		if c.MustGet("username").(string) != username {
+			log.Error("auth error: unauthorized")
+			c.JSON(utils.ErrorStatusCodes[utils.ErrUnauthorized],
+				presenter.CreateErrorResponse(utils.ErrUnauthorized))
+			return
+		}
+
+		user, err := service.FindUserByUsername(username)
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrUserNotFound], presenter.CreateErrorResponse(utils.ErrUserNotFound))
+			return
+		}
+
+		outputUser := user.GetUserWithProject()
+
+		projects, err := service.GetProjectsByUserID(outputUser.ID, false)
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+
+		outputUser.Projects = projects
+
+		c.JSON(200, gin.H{"data": outputUser})
+	}
+}
+
+// GetProject queries the project with a given projectID from the database
+func GetProject(service services.ApplicationService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		projectID := c.Param("project_id")
+
+		err := validations.RbacValidator(c.MustGet("uid").(string), projectID,
+			validations.MutationRbacRules["getProject"], string(entities.AcceptedInvitation), service)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrUnauthorized],
+				presenter.CreateErrorResponse(utils.ErrUnauthorized))
+			return
+		}
+
+		project, err := service.GetProjectByProjectID(projectID)
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+
+		// Fetching user ids of all the members in the project
+		var uids []string
+		for _, member := range project.Members {
+			uids = append(uids, member.UserID)
+		}
+
+		authUsers, err := service.FindUsersByUID(uids)
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+
+		memberMap := make(map[string]entities.User)
+		for _, authUser := range *authUsers {
+			memberMap[authUser.ID] = authUser
+		}
+
+		var members []*types.Member
+
+		// Adding additional details of project members
+		for _, member := range project.Members {
+			members = append(members, &types.Member{
+				UserID:        memberMap[member.UserID].ID,
+				UserName:      memberMap[member.UserID].UserName,
+				Name:          memberMap[member.UserID].Name,
+				Role:          member.Role,
+				Email:         memberMap[member.UserID].Email,
+				Invitation:    member.Invitation,
+				JoinedAt:      member.JoinedAt,
+				DeactivatedAt: memberMap[member.UserID].DeactivatedAt,
+			})
+		}
+
+		if err != nil {
+			return
+		}
```

Comment:
We can remove this?

Reply:
Yes @Jonsy13 thanks for pointing out, I will remove this